### PR TITLE
chore(docs): add migration guide sections for storage and notifications

### PR DESCRIPTION
**Pull request by esauerbo**
#### Description of changes
Added sections in migration guide for `ui-react-storage` and `ui-react-notifications` 

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx [hunk 1]
```diff
@@ -733,3 +697,88 @@ Optionally update your App with the new prop usage:
 #### CDN CSP Policy
 
 The 2.x version of the `FaceLivenessDetector` has updated the default CDN for TensorFlow and Blazeface to a new URL. If your application has an existing CSP policy, update your policy to allow https://cdn.liveness.rekognition.amazonaws.com. Please look over the [Liveness Config](../connected-components/liveness/customization#facelivenessdetectorconfig) section for more information.
+
+
+## `@aws-amplify/ui-react-notifications`
```
Comment (calebpollman) [inline docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:702]: Concerned abt the discoverability of the migration changes with moving `@aws-amplify/ui-react-notifications` and `@aws-amplify/ui-react-storage` so far down in the guide?

Comment (esauerbo) [reply docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:702]: I could add a note in the react section that points you to them? I just went alphabetically

Comment (calebpollman) [reply docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:702]: Adding package links at the top of the page would make sense imo for at least the short term. @dbanksdesign You have any thoughts?

Comment (reesscot) [reply docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:702]: Yeah, I like the idea of package links, though I suppose they are in the TOC?

Comment (dbanksdesign) [reply docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:702]: Yup h2s and h3s are in the TOC, package names being h2s they are in there: https://ui.docs.amplify.aws/react/getting-started/migration

CleanShot 2023-11-15 at 20 51 10@2x

If we wanted we could add a list of links with package names in the body?

**Review by dbanksdesign — DISMISSED** [review]

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx (hunk 1 above)
Comment (calebpollman) [reply docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:702]: > If we wanted we could add a list of links with package names in the body?

This is what i was thinking, imo would improve discoveravbility

**Review by reesscot — DISMISSED** [review]

**Review by calebpollman — DISMISSED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**